src/js/controller/controller.js

```diff
@@ -539,7 +539,9 @@ Object.assign(Controller.prototype, {
             }
 
             // It wasn't the last item in the playlist,
-            //  so go to the next one
+            //  so go to the next one and trigger an autoadvance event
+            const related = _api.getPlugin('related');
+            triggerAdvanceEvent(related, 'nextAutoAdvance');
             _next({ reason: 'playlist' });
```

Comment:
Doesn't `reason: 'playlist'` imply an auto advance?

Reply:
Yeah but if I understand the setup correctly, the next event that would fire would be a `beforePlay` when the playlist item has already changed. This autoadvance event needs to track before the playlist switches, so triggering a separate event ensures we tracking the autoadvance at the correct time. See `feedClick` vs `relatedClick`

Reply:
Gotcha

Reply:
Is it critical that we track normal playlist progression? What about "beforePlaylistItem" for any playlist item change to track drop off? I'm OK with this but am very concerned that we're overloading the player with events for something that we want 'related' to track.